tests/test_engine.py

```diff
@@ -357,6 +369,46 @@ def test_close_engine_spiders_downloader(self):
         self.assertEqual(len(e.open_spiders), 0)
 
 
+class CancelledRequestEngineTest(EngineTest):
+
+    @defer.inlineCallbacks
+    def test_crawler(self):
+        for spider in TestSpider, DictItemsSpider:
+            self.run = CancelledRequestCrawlerRun(spider)
+            with LogCapture() as log:
+                yield self.run.run()
+                log.check_present(("scrapy.core.downloader.handlers.http11",
+                                   "DEBUG",
+                                   "Cancelled request <GET http://localhost:{}/redirected> from signal handler"
+                                   " CancelledRequestCrawlerRun.bytes_received".format(self.run.portno)))
+                log.check_present(("scrapy.core.downloader.handlers.http11",
+                                   "DEBUG",
+                                   "Cancelled request <GET http://localhost:{}/> from signal handler"
+                                   " CancelledRequestCrawlerRun.bytes_received".format(self.run.portno)))
+                log.check_present(("scrapy.core.downloader.handlers.http11",
+                                   "DEBUG",
+                                   "Cancelled request <GET http://localhost:{}/numbers> from signal handler"
+                                   " CancelledRequestCrawlerRun.bytes_received".format(self.run.portno)))
+            self._assert_visited_urls()
+            self._assert_scheduled_requests(urls_to_visit=9)
+            self._assert_downloaded_responses()
+            self._assert_signals_caught()
+            self._assert_bytes_received()
+
+    def _assert_bytes_received(self):
+        self.assertEqual(9, len(self.run.bytes))
+        for request, data in self.run.bytes.items():
+            joined_data = b"".join(data)
+            self.assertTrue(len(data) == 1)  # signal was fired only once
+            if self.run.getpath(request.url) == "/numbers":
+                self.assertTrue(len(data) == 1)  # signal was fired only once
```

Comment:
This seems redundant, given the line two lines above.

Reply:
Oh right, forgot to remove this one :+1: